### PR TITLE
TG's Got the Pop Factor: The Singing Update

### PR DESCRIPTION
**Pull request by EOBGames**
## About The Pull Request
Adds singing as a new speech verb. Simply use % (the same as you would use # for whispering) to start belting out a tune! This applies a special span class, speech verb, and adds music notes to your message for optimum lyrical loveliness.
Example image:
wellthankgodthatsover

## Why It's Good For The Game
A good sing along is good for crew morale.

## Changelog
:cl:
add: The crew has finally remembered how to sing. Simply use % to start singing your favourite tune!
/:cl:

### CODE REVIEW

**Comment by Anonmare** [comment]
Now I can finally host that talent competition 

Welcome ladies and Gentlemen to the first annual Nanotrasen Top of the Pops. First up, we have the Cargo Shop Quartet followed by our resident Mime "Deafening Silence". Later please enjoy the musical stylings of Bubbles the Talking Monkey Clown.

File: code/__DEFINES/logging.dm [hunk 1]
```diff
@@ -37,6 +37,7 @@
 #define LOG_VIRUS			(1 << 16)
 #define LOG_CLONING			(1 << 17)
 #define LOG_SHUTTLE			(1 << 18)
+#define LOG_SING            (1 << 19)
```
Comment (AnturK) [inline code/__DEFINES/logging.dm]: Absolutely no need to separate this log from say

File: code/modules/goonchat/browserassets/css/browserOutput.css [hunk 2]
```diff
@@ -366,6 +366,7 @@ em						{font-style: normal;	font-weight: bold;}
 .greentext				{color: #059223;	font-size: 185%;}
 .redtext				{color: #c51e1e;	font-size: 185%;}
 .clown					{color: #ff70c1;	font-size: 160%;	font-family: "Comic Sans MS", cursive, sans-serif;	font-weight: bold;}
+.singing				{font-family: "Apple Chancery", cursive, sans-serif; font-style: italic;}
```
Comment (AnturK) [inline code/modules/goonchat/browserassets/css/browserOutput.css]: Chances of users having that specific font are kinda slim.

File: code/modules/mob/living/say.dm [hunk 3]
```diff
@@ -196,6 +196,11 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	if(language)
 		var/datum/language/L = GLOB.language_datum_instances[language]
 		spans |= L.spans
+	
+	if(message_mode == MODE_SING)
+		var/randomnote = pick("♩", "♪", "♫")
```
Comment (AnturK) [inline code/modules/mob/living/say.dm]: Use \uXXXX macro, we're still supporting 512 compilation.

**Comment by MrDoomBringer** [comment]
image
feature incorporation when?

File: code/modules/mob/say.dm [hunk 4]
```diff
@@ -36,6 +36,19 @@
...
+	set name = "Sing"
+	set category = "IC"
+	if(GLOB.say_disabled)	//This is here to try to identify lag problems
+		to_chat(usr, "<span class='danger'>Speech is currently admin-disabled.</span>")
+		return
+	sing(message)
+
+///sing a lovely song
+/mob/proc/sing(message, datum/language/language=null)
+	say(message, language) //only living mobs can sing, dead things are pretty tonedeaf
```
Comment (AnturK) [inline code/modules/mob/say.dm]: This does nothing but say. Completely unnecessary.

**Review by AnturK — CHANGES_REQUESTED** [review]

**Comment by Shadowflame909** [comment]
> image
> feature incorporation when?
> 
> Also vending machines should sing their slogans
> Also elvis mutation should make you perma-sing

Wearing vocaloid outfits changes your singing color to the costume color.

To simulate the simulation of song

**Comment by ExcessiveUseOfCobblestone** [comment]
Nya~ I mean 🎵 Nya

**Comment by XDTM** [comment]
Some keyboards don't have the ~ symbol; is there a more international alternative?

**Comment by HomocidlePancakes** [comment]
Even my Russian keyboard has a tilde, even though it’s not used in writing. Fairly certain a good chunk of keyboards has a tilde too.

**Comment by HomocidlePancakes** [comment]
It’s supposed to be above tab.

**Comment by XDTM** [comment]
Must be an Italian thing, then. Doesn't affect me personally since i have an english layout, but it might happen to others.

**Comment by Tlaltecuhtli** [comment]
i dont have stinky line symbol, how about §

**Comment by Grazyn** [comment]
I'll just press alt+126

**Comment by XDTM** [comment]
Could also be % or &, i think they're unused

**Comment by trollbreeder** [comment]
Why not microphones that boost your voice to loud when you sing?

**Comment by gitbirb** [comment]
Good update, cuban petes can finally sing to the drum beat. Also i suggest making science labcoats as a result give you a unique singing color to symbolize it.

**Comment by SweptWasTaken** [comment]
When can we have Starbound style synced up singalongs?

**Comment by ATH1909** [comment]
> Good update, cuban petes can finally sing to the rumba beat. Also i suggest making science labcoats as a result give you a unique singing color to symbolize it.

Eh, I think it'd be better to tie that to sombreros and/or the Cuban Pete hat from the gene vault away mission.

**Pull request merged**